File: web/src/main/java/top/bettercode/summer/web/serializer/CentDeserializer.java
package top.bettercode.summer.web.serializer;

import com.fasterxml.jackson.core.JsonParser;
import com.fasterxml.jackson.databind.DeserializationContext;
import com.fasterxml.jackson.databind.JsonDeserializer;
import java.io.IOException;
import top.bettercode.summer.tools.lang.util.MoneyUtil;

/**
 * @author Peter Wu
 */
public class CentDeserializer extends JsonDeserializer<Long> {

  @Override
  public Long deserialize(JsonParser p, DeserializationContext ctxt)
      throws IOException {
    return MoneyUtil.toCent(p.getValueAsString());
  }
}

File: web/src/main/kotlin/top/bettercode/summer/web/serializer/CentDeserializer.kt
package top.bettercode.summer.web.serializer

import com.fasterxml.jackson.core.JsonParser
import com.fasterxml.jackson.databind.DeserializationContext
import com.fasterxml.jackson.databind.JsonDeserializer
import top.bettercode.summer.tools.lang.util.MoneyUtil.toCent

/**
 * @author Peter Wu
 */
class CentDeserializer : JsonDeserializer<Long>() {
    override fun deserialize(p: JsonParser, ctxt: DeserializationContext): Long {
        return toCent(p.valueAsString)
    }
}
